services/test_analytics/ta_process_flakes.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from datetime import datetime
+
+from django.db import transaction
+from django.db.models import Q, QuerySet
+from redis.exceptions import LockError
+from shared.django_apps.reports.models import CommitReport, ReportSession
+from shared.django_apps.test_analytics.models import Flake
+from shared.django_apps.timeseries.models import Testrun
+
+from services.redis import get_redis_connection
+
+log = logging.getLogger(__name__)
+
+FAIL_FILTER = Q(outcome="failure") | Q(outcome="flaky_failure") | Q(outcome="error")
+
+LOCK_NAME = "ta_flake_lock:{}"
+KEY_NAME = "ta_flake_key:{}"
+
+
+def get_relevant_uploads(repo_id: int, commit_id: str) -> QuerySet[ReportSession]:
+    return ReportSession.objects.filter(
+        report__report_type=CommitReport.ReportType.TEST_RESULTS.value,
+        report__commit__repository__repoid=repo_id,
+        report__commit__commitid=commit_id,
+        state__in=["processed"],
+    )
+
+
+def fetch_current_flakes(repo_id: int) -> dict[bytes, Flake]:
+    return {
+        bytes(flake.test_id): flake for flake in Flake.objects.filter(repoid=repo_id)
+    }
+
+
+def get_testruns(
+    upload: ReportSession, curr_flakes: dict[bytes, Flake]
+) -> QuerySet[Testrun]:
+    upload_filter = Q(upload_id=upload.id)
+    flaky_pass_filter = Q(outcome="pass") & Q(test_id__in=curr_flakes.keys())
+    return Testrun.objects.filter(upload_filter & (FAIL_FILTER | flaky_pass_filter))
+
+
+def handle_pass(curr_flakes: dict[bytes, Flake], test_id: bytes):
+    curr_flakes[test_id].recent_passes_count += 1
+    curr_flakes[test_id].count += 1
+    if curr_flakes[test_id].recent_passes_count == 30:
```

Comment:
moving this magic constant to the top level would make sense. and an explanation of "after X passes in a row, the test is not marked as flaky anymore"